### DISCUSSION TITLE
Is this a bug or a misundestanding/misconfiguration on my part?

### DISCUSSION THREAD

**Discussion by muelthom** [vscode-neovim/vscode-neovim#1278] [Q&A] [answered]
Before opening an issue, I first wanted to ask whether this is a known behavior...

I have this very strange bug/behavior in VS Code; the syntax colors are (partly) taken from my ordinary Neovim configuration, although that should not be the case. Also a strange bullet symbol • is shown at the and or start of some lines.
I have recorded a short [video](https://nx10546.your-storageshare.de/s/CDMydnXmLaA2gqX) because that shows the bug quite well.

More background info: my ordinary Neovim configuration uses the [Rosé Pine colorscheme](https://github.com/rose-pine/neovim) and a couple of plugins. For the _vscode-neovim_ extension I've created a dedicated `vscode-init.lua` Lua file (see below) to prevent any plugins to load that could cause issues with the VS Code plugin. For reference, the same file as shown in the video looks like [this](https://nx10546.your-storageshare.de/s/4b5owgX6Z4LisER) in Neovim (I added some print statements to check whether the file was loaded at all).

```lua
if vim.g.vscode then
    -- VSCode extension
    print("Loaded vscode-init.lua")
else
    -- Ordinary Neovim
    print("This should not be called")
end
```

My ordinary Neovim `init.lua` file looks as follows:
```lua
if vim.g.vscode then
    -- VSCode extension
else
    -- Ordinary Neovim

    -- File explorer (nvim-tree) settings.
    -- It is strongly recommended to disable netrw. As it is a bundled plugin it
    -- must be disabled manually at the start of the "init.lua".
    vim.g.loaded_netrw = 1
    vim.g.loaded_netrwPlugin = 1

    -- Custom keybindings
    require("remaps")

    -- Editor settings
    require("settings")

    -- Configure plugins
    require("packer-plugin")
end
```

What could cause this behavior? Is there something I did wrong, or I misunderstand? I'd appreciate any help/hint.

If I start the plugin using the `--clean` option, everything is fine. However, then I do not have my key remaps :(

**Comment by theol0403** [comment, marked as answer]
Packer adds plugins to your runtimepath, disabling the plugin spec does not prevent it from running.

Use `cond`, or use lazyvim with vscode spec.

**Reply by muelthom** [reply]
Oh :( Thank you for the clarification Theo!

**Reply by muelthom** [reply]
For others that might also be interested in this topic: https://github.com/wbthomason/packer.nvim/issues/933
